editor/components/post-permalink/index.js

```diff
@@ -72,8 +168,14 @@ export default connect(
 	( state ) => {
 		return {
 			isNew: isEditedPostNew( state ),
-			link: getEditedPostAttribute( state, 'link' ),
+			previewLink: getEditedPostPreviewLink( state ),
+			samplePermalinkData: getEditedPostAttribute( state, 'sample_permalink' ),
+
+			permalinkStructure: window.wpApiSettings.schema.permalink_structure,
 		};
+	},
+	{
+		editPost,
```

Comment:
This is fine enough since we already have `connect` here anyways, but new usage is encouraged (and hopefully simplified) to use [`withDispatch`](https://github.com/WordPress/gutenberg/tree/master/data#withdispatch-mapdispatchtoprops-function--function) instead:

```js
import { withDispatch } from '@wordpress/data';
import { Component, compose } from '@wordpress/element';

// ...

export default compose( [
	connect( ( state ) => {
		return {
			isNew: isEditedPostNew( state ),
			previewLink: getEditedPostPreviewLink( state ),
			samplePermalinkData: getEditedPostAttribute( state, 'sample_permalink' ),

			permalinkStructure: window.wpApiSettings.schema.permalink_structure,
		};
	} ),
	withDispatch( ( dispatch ) => {
		const { editPost } = dispatch( 'core/editor' );
		return { editPost };
	} ),
] )( PostPermalink );
```

Reply:
👍🏻